### PR TITLE
Add new indentation_width rule

### PR DESCRIPTION
**Pull request by fredpi**
It's been a while since https://github.com/realm/SwiftLint/issues/227 has last been discussed and it seems that an advanced indentation rule is somehow also pending some clarification on Xcode's side. However, since this issue has been open for such a long time, I thought about adding a basic additive indentation rule that only does the following things:

- Check whether the next line is on the same level as the previous line or one tab or `n` spaces indented (where `n` can be configured via `indentation_width`) or unindented keeping the grid.
- Check whether the first line is not indented.
- Check whether spaces and tabs aren't mixed up in one line's indentation.

This PR implements such a basic rule. I tested it with a large project we have in our company and found out that it matches the indentation style we already pursued before:

```
let testStruct = TestStruct(
    name: "Test",
    indentation: "Cool"
)
```

However, this rule won't be compatible with other indentation styles like the one Xcode does by default and that is enforced by the `vertical_parameter_alignment_on_call` opt-in-rule:

```
let testStruct = TestStruct(name: "Test",
                            indentation: "Cool")
```

That's why this is an opt-in rule. As such, it won't do harm to those following another indentation style, yet enable built-in indentation linting for those who follow this basic additive pattern. Previously, we had implemented such a rule using multiple regexes that checked for indentation but didn't always work properly, so having this built-in is definitely a plus.

I want to mention that, according to this rule, unindentation can not only be done by one step (~ `indentation_width` spaces), but also by multiples of one step. This is to allow for the following indentation which I consider valid:

```
if
    let myLet = someBool ?
        myLet1 :
        myLet2
{
    // Do something with myLet
} 
```

### CODE REVIEW

**Comment by fredpi** [comment]
Any thoughts on this? ;)

**Comment by fredpi** [comment]
As I'm personally not that active in reviewing other people's PRs, I'm certainly not in the position to complain that, seemingly, nobody wants to review this.

However, I'd still like to see this merged at some point. Don't be scared by the `41903 Warnings` message, it's because not all projects follow the indentation pattern this rule enforces.

**Comment by fredpi** [comment]
@jpsim Would you mind having a look here? ;)

**Comment by fredpi** [comment]
Anyone?

**Comment by fredpi** [comment]
Rebasing this every 3 weeks or so is getting frustrating...

If nobody has time to review this, maybe one could give me push access so that I can self-review and merge thereafter...
As this is just an opt-in-rule, it doesn't do any harm...

**Comment by fredpi** [comment]
`/__w/1/s/Tests/SwiftLintFrameworkTests/LinterCacheTests.swift:351: error: LinterCacheTests.testDetectSwiftVersion : XCTAssertEqual failed: ("5.1.3") is not equal to ("5.1.2")`

This test failure does not seem to be related to this PR specifically...

**Comment by fredpi** [comment]
@marcelofabri Would you mind having a look here?

**Comment by jpsim** [comment]
> Rebasing this every 3 weeks or so is getting frustrating...
> 
> If nobody has time to review this, maybe one could give me push access so that I can self-review and merge thereafter...
> As this is just an opt-in-rule, it doesn't do any harm...

Sorry for the lack of communication here @fredpi. I don't have a whole lot of time for OSS work lately 😞

I've given you commit access to the repo. In general, you should feel empowered to merge PRs to SwiftLint if other maintainers aren't being responsive.

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,8 +9,11 @@
 * None.
 
 #### Enhancements
-
-* None.
+  
+* Add new indentation opt-in-rule checking for
```
Comment (jpsim) [inline CHANGELOG.md]: Nit:
```suggestion
* Add new indentation opt-in rule checking for
```

**Pull request merged**